telemetrylib/SQL.cpp

```diff
@@ -0,0 +1,151 @@
+
+        // TODO SPLIT THIS UP INTO TWO PARTS:
+        //        X 1. table-name: ("_" + first_msec, which is equivalent to the response from the "/add-table/*" request) table name created from the start time of the session
+        //        X 2. dataset-time: (curr_msec) timestamp associated with the byte array being sent to the server
+        // TODO request.setHeader(QNetworkRequest::ContentDispositionHeader, QVariant("identifier; session-time=" + QString::fromStdString(std::to_string(first_msec))));
+
+
+        //request.setHeader(QNetworkRequest::ContentDispositionHeader, QVariant("identifier; table-name=" + tableName + ",dataset-time=" + QString::fromStdString(std::to_string(curr_msec))));
+        // TODO request.setHeader(QNetworkRequest::ContentDispositionHeader, QVariant("identifier; session-time=" + QString::fromStdString(std::to_string(first_msec)) + ",dataset-time=" + QString::fromStdString(std::to_string(curr_msec))));
+        // TODO request.setHeader(QNetworkRequest::ContentDispositionHeader, QVariant("identifier; session-time=_quickTest,dataset-time=" + QString::fromStdString(std::to_string(curr_msec))));
+        // TODO request.setHeader(QNetworkRequest::ContentDispositionHeader, QVariant("identifier; session-time=" + QString::fromStdString(std::to_string(curr_msec))));
+
+        // TODO Note in case message frequency becomes an issue: Set up a message queue for HTTP requests,
+        //      and send groups of byte arrays to the server at less frequent intervals (/as soon as we can send our next message)
+
+        // TODO Remove: qDebug() << this->thread() << "\t\t" << restclient->thread() << "\t\t" << sec_counter;
+
+        // TODO Allows HTTP pipelining so that the request doesn't wait for a response from the server before allowing a new message to be sent
+        //      This should help to avoid stalling while waiting for a response from the server
+        //      NOTE - The number of messages sent before a response from the first message is received might still be limited
+        // TODO Send the byte array, along with the corresponding timestamp, to the server
+        // TODO reply = this->restclient->post(request, bytes);
+        this->restclient->post(request, bytes);
+    }
+
+    std::string receiveData() override{
+        QJsonObject json = QJsonDocument::fromJson(reply->readAll()).object();
+        //QJsonObject json = QJsonDocument::fromJson(QByteArray::fromStdString("JSON Derulo")).object();
+
+        if (json.isEmpty()) {
+            qDebug() << "EMPTY JSON";
+            return "nada";
+        } else {
+            tableName = json.take("response").toString();
+            qDebug() << "HTTP response (table name): " << tableName;
+            return tableName.toStdString();
+            // TODO Automatically delete server responses since they aren't used after reading the table name
+            this->restclient->setAutoDeleteReplies(true);
```

Comment:
It looks like this isn't what's being used to set the table name (readReply() is used), so setting tableName can be removed or replaced with a more generic code for reading a response.

Reply:
See comment in library.cpp, but it looks like this function isn't utilized, so it could be removed entirely

Reply:
commented out